src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsReceiver.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\AmazonSqs\Transport;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\LogicException;
+use Symfony\Component\Messenger\Exception\MessageDecodingFailedException;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Transport\Receiver\MessageCountAwareInterface;
+use Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface;
+use Symfony\Component\Messenger\Transport\Serialization\PhpSerializer;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Contracts\HttpClient\Exception\HttpExceptionInterface;
+
+/**
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class AmazonSqsReceiver implements ReceiverInterface, MessageCountAwareInterface
+{
+    private $connection;
+    private $serializer;
+
+    public function __construct(Connection $connection, SerializerInterface $serializer = null)
+    {
+        $this->connection = $connection;
+        $this->serializer = $serializer ?? new PhpSerializer();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get(): iterable
+    {
+        try {
+            $sqsEnvelope = $this->connection->get();
+        } catch (HttpExceptionInterface $e) {
+            throw new TransportException($e->getMessage(), 0, $e);
+        }
+        if (null === $sqsEnvelope) {
+            return;
+        }
+
+        try {
+            $envelope = $this->serializer->decode([
+                'body' => $sqsEnvelope['body'],
+                'headers' => $sqsEnvelope['headers'],
+            ]);
+        } catch (MessageDecodingFailedException $exception) {
+            $this->connection->delete($sqsEnvelope['id']);
+
+            throw $exception;
+        }
+
+        yield $envelope->with(new AmazonSqsReceivedStamp($sqsEnvelope['id']));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function ack(Envelope $envelope): void
+    {
+        try {
+            $this->connection->delete($this->findSqsReceivedStamp($envelope)->getId());
+        } catch (HttpExceptionInterface $e) {
+            throw new TransportException($e->getMessage(), 0, $e);
```

Comment:
Are we okey with `Symfony\Component\Messenger\Exception\LogicException` bubbling up?

Reply:
hmm.. IMHO this is more a Runtime Exception (this is what other transport do).

Maybe if we want to be more accurate:
```
        try {
            ...
        } catch (ServerExceptionInterface $e) {
            throw new TransportException($e->getMessage(), 0, $e);
        } catch (HttpExceptionInterface $e) {
            throw new LogicException($e->getMessage(), 0, $e);
        }
```
whats do you think?

Reply:
I meant that the `findSqsReceivedStamp` could throw a LogicException. We are not catching it and wrapping it in a `TransportException`.

Reply:
hmm.. I'm not sure to understand your point. If the method ack is called with an enveloppe without SQS stamp, it seems normal to me to let the LogicException bubbling.

note: AMQP and Redis use the same logic: https://github.com/symfony/symfony/blob/524ee7acb676d3e2109c8f4afb8e9fa60711ee4f/src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpReceiver.php#L83-L92